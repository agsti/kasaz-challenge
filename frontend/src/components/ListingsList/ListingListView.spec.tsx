import React from 'react';
import { render, mount } from 'enzyme';
import { describe, expect, it } from '@jest/globals';

import 'jsdom-global/register';

import ListingListView from './ListingsListView';

const onScroll = jest.fn()
const component = mount(<ListingListView
    listings={[
        {
            nBath :1,
            nRooms: 2,
            picture: "/img.jpg",
            price: 1000000,
            priceSqm: 312.12,
            sqm: 500,
            title: "title"
        }
    ]}
    scrollPosition={500}
    onScroll={onScroll}
/>);

describe('ListingListView', () => {
    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });


    it('Scrolling should fire onScroll', () =>{
        component.simulate('scroll', { deltaY: 5000 });

        expect(onScroll).toHaveBeenCalled()
    });

    it('ScrollPosition sets div scroll', () =>{
        expect(component.getDOMNode().scrollTop).toBe(500);
    });
});
