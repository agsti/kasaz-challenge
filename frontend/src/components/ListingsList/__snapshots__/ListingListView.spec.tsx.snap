// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingListView should render correctly 1`] = `
<ListingListView
  listings={
    Array [
      Object {
        "nBath": 1,
        "nRooms": 2,
        "picture": "/img.jpg",
        "price": 1000000,
        "priceSqm": 312.12,
        "sqm": 500,
        "title": "title",
      },
    ]
  }
  onScroll={[MockFunction]}
  scrollPosition={500}
>
  <div
    className="listing-list"
    onScroll={[Function]}
  >
    <ListingItem
      key="0"
      nBath={1}
      nRooms={2}
      picture="/img.jpg"
      price={1000000}
      priceSqm={312.12}
      sqm={500}
      title="title"
    >
      <div
        className="listing-container"
      >
        <div
          className="listing-image-price"
        >
          <img
            src="/img.jpg"
          />
          <div
            className="listing-prices"
          >
            <div
              className="listing-price"
            >
              1,000,000
               €
            </div>
            <div
              className="listing-price-sqm"
            >
              312
               €/m²
            </div>
          </div>
        </div>
        <div
          className="listing-title"
        >
          title
        </div>
        <div
          className="listing-info-container"
        >
          <div
            className="listing-info"
          >
            <img
              src=""
            />
            500
             m²
          </div>
          <div
            className="listing-info"
          >
            <img
              src=""
            />
            2
             habs.
          </div>
          <div
            className="listing-info"
          >
            <img
              src=""
            />
            1
             baños
          </div>
        </div>
      </div>
    </ListingItem>
  </div>
</ListingListView>
`;
